File: lab4/queue.py
class Item:
    def __init__(self, x):
        self.data = x
        self.next = None
class LList:
    def __init__(self):
        self.start = None
    def ins(self, data):
        new_item = Item(data)
        if self.start is None:
            self.start = new_item
            return
        hold = self.start
        while hold.next is not None:
            hold = hold.next
        hold.next = new_item

    def pop(self):
        while self.start is not None:
            print(self.start.data)
            self.start = self.start.next
coolqueue = LList()
while(True):
    a = input("Enter int value (0 to exit):")
    if a == 0:
        break
    coolqueue.ins(a)
coolqueue.pop()

File: lab4/queue.c
#include <stdlib.h>
#include <stdio.h>

struct item {
    int x;
    struct item *next;
};

int main(){
    struct item *coolqueue = NULL, *hold, *inpq, *nlast;
    int stop = 0, elements = 0, inp;
    do{
        printf("Enter int element to insert (0 to exit): ");
        scanf("%d", &inp);
        if(inp != 0){
            elements++;
            inpq = (struct item*)malloc(sizeof(struct item));
            inpq->x = inp;
            inpq->next = NULL;
            if(coolqueue == NULL){
                coolqueue = inpq;
            }
            else{
                hold = coolqueue;
                while(hold->next != NULL){
                    hold = hold->next;
                }
                hold->next = inpq;
            }
        }
        else{
            stop = 1;
        }
    }while(stop == 0);
    while(elements > 0 && coolqueue != NULL){
        elements--;
        if(coolqueue->next == NULL){
            inp = coolqueue->x;
            printf("POPPING: %d\n", inp);
            free(coolqueue);
            coolqueue = NULL;
            printf("QUEUE EMPTY!\n");
        }
        else{
            hold = coolqueue;
            inp = coolqueue->x;
            coolqueue = coolqueue->next;
            printf("POPPING: %d\n", inp);
            free(hold);
        }
    }
}